### ISSUE TITLE
lmdb: finalizer for Txn

### ISSUE THREAD

**Issue by bmatsuo** [bmatsuo/lmdb-go#27]
Relates to #20 

A finalizer for Txn should only be required on unmanaged Txn types. Because managed Txn types have
a deferred Abort operation their resources will always be released.

Adequate benchmarks for unmanaged Txn objects must be in place before implementation and `benchcmp` should verify the performance overhead of a finalizer. Further, adequate benchmarks of Txn.Reset and Txn.Renew must be in place so that their benefits can be further quantified and compared.
